plugin/storage/integration/elasticsearch_test.go

```diff
@@ -105,14 +102,40 @@ func (s *ESStorageIntegration) initializeES(t *testing.T, allTagsAsFields bool)
 		elastic.SetSniff(false))
 	require.NoError(t, err)
 	s.logger, _ = testutils.NewLogger()
-
 	s.client = rawClient
 	s.v8Client, err = elasticsearch8.NewClient(elasticsearch8.Config{
 		Addresses:            []string{queryURL},
 		DiscoverNodesOnStart: false,
 	})
 	require.NoError(t, err)
 
+	// Initialize ES Factory
+	cfg := es.NewFactory().Options.Primary.Configuration
```

Comment:
why do you need to go through NewFactory? The factory is leaked this way. What if you just instantiate the config struct directly? Are you relying on some initialization happening in NewFactory?

Reply:
I did initially initialise config struct directly in the below manner. Many tests were failing, so I presumed the default initialization would not have been occurring, that's why thought of initializing using NewFactory, and change the Fields manually.

Reply:
Done! Didn't need the NewFactory, initialized it using NewOptions struct.

Reply:
Hi, I am not able to debug why the GetOperations test is failing.
I have went through the initialisation of GetOperations and used Debugger, but "Actual" is always being initialised as nil.
Is it due to a difference in the Client Fn?
I would really appreciate your insight..

Reply:
My recommendation is to add logging of the query that we send to ES (a JSON). You can then run this query directly against ES and troubleshoot why it does not return the data.

Reply:
Yes, I did that. The Example_Trafe query is being sent to the loadParseandWrite method successfully, I e writing the trace to ES but it's failing (Returning nil) while fetching it from the SpanReader

Reply:
I was referring to logging reads, not writes. I see that ES client has this logger that is used to log requests:
```
	tracelog                  Logger       // trace log for debugging
```
it can be set via SetTraceLog option when creating the client

Reply:
Got it, will try this.
Thank you!